pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -630,7 +630,7 @@ func (c *Reconciler) runNextSchedulableTask(ctx context.Context, pr *v1beta1.Pip
 	}
 
 	for _, rprt := range nextRprts {
-		if rprt == nil || rprt.Skip(pipelineRunFacts) {
+		if rprt == nil || rprt.Skip(pipelineRunFacts) || rprt.IsFinallySkipped(pipelineRunFacts) {
```

Comment:
hey @jerop, based on my understanding, `rprt.Skip(pipelineRunFacts)` returns `false` for any finally task but `rprt.IsFinallySkipped(pipelineRunFacts)` evaluates when expression and returns `true` since the when expression evaluates to `false`.

Reply:
but this would work, provided `nextRprts` contains the finally tasks 🤣